.github/workflows/automated-tests.yml

```diff
@@ -14,9 +14,9 @@ jobs:
         runs-on: ubuntu-latest
         steps:
           - name: Checkout repo
-            uses: actions/checkout@v3
+            uses: actions/checkout@v4
```

Comment:
What does this have to do with the rest?

Reply:
I received a deprecation warning, and was asked to use a newer version of these 2 actions.

It is unrelated, so strictly speaking, I should have done another PR for it. But because it's such a small thing, I decided to make 1 PR for both.